spec/models/challenge_spec.rb

```diff
@@ -1,11 +1,83 @@
 require "rails_helper"
 
 RSpec.describe Challenge, type: :model do
-  context "create a challenge" do
-    it "A challenge must have a title and details" do
-      user = create(:user)
-      challenge = build(:challenge, user: user)
-      expect(challenge).to be_valid
+  describe "validations" do
+    it { should validate_presence_of(:user) }
+  end
+
+  describe "associations" do
+    it { should belong_to(:user).without_validating_presence }
+    it { should have_many(:path_challenges) }
+    it { should have_many(:pathways).through :path_challenges }
+  end
+
+  describe "create a challenge" do
+    subject {
+      described_class.new(
+        user: create(:user),
+        title: "Roman numeral to integer",
+        details: "Convert Roman numerals to numbers and convert numbers to Roman numerals"
+      )
+    }
+
+    it "is valid with valid attributes" do
+      expect(subject).to be_valid
+    end
+
+    describe "details validation" do
+      context "no details" do
+        it "is not valid without details" do
+          subject.details = nil
+          expect(subject).to_not be_valid
+        end
+      end
+
+      context "details are too short" do
+        it "is not valid with too short details" do
+          subject.details = "abc"
+          expect(subject).to_not be_valid
+        end
+      end
+
+      context "details too long" do
+        it "is not valid with too long details" do
+          subject.details = ("a" * 1501).to_s
+          expect(subject).to_not be_valid
+        end
+      end
+    end
+
+    describe "title validation" do
+      context "no title" do
+        it "is not valid without a title" do
+          subject.title = nil
+          expect(subject).to_not be_valid
+        end
+      end
+
+      context "title too short" do
+        it "is not valid with too short a title" do
+          subject.title = "abc"
+          expect(subject).to_not be_valid
+        end
+      end
+
+      context "title too long" do
+        it "is not valid with too long a title" do
+          subject.title = ("a" * 51).to_s
+          expect(subject).to_not be_valid
+        end
+      end
+
+      context "invalid chars" do
+        it "is not valid with invalid characters" do
+          subject.title = "!*?!#@"
+          expect(subject).to_not be_valid
```

Comment:
This could also be invalid because it's less than 8 characters

Reply:
with invalid tests, you could do .save and expect().to raise_error() and put details of the expected error if you want to be more specific

Reply:
> This could also be invalid because it's less than 8 characters

nice catch. I'll also look into expect and raise error tomorrow, thanks!

Reply:
> This could also be invalid because it's less than 8 characters

nice catch. I'll also look into expect and raise error tomorrow, thanks!

Reply:
nice looks

Reply:
used raise_error for specific errors expected; is this what you were after @CraigNorford?